.travis.yml

```diff
@@ -31,24 +36,44 @@ install:
         done
     }
 
+  - |
+    # Run clippy unless $CLIPPY_RUSTFMT is set to "no". If it is set to
+    # "if_available", then this is skipped if clippy is not available. With
+    # "yes", unavailability counts as an error.
+    run_clippy() {
+        if [ "x$CLIPPY_RUSTFMT" != "xno" ]; then
```

Comment:
Out of curiosity, what's the reason for prefixing env values with a `x`?

Reply:
It makes completely no sense here.

Once upon a time, I read something about empty arguments. `if [ $FOO = bar ]` would expand to `if [ = bar ]` when `$FOO` is empty and that would cause an error in non-bash (or something like that). Once upon a time, I read something like this and basically cargo-culted the extra `x` around.

Reply:
Google says its even totally unnecessary here, if I understand this correctly: https://unix.stackexchange.com/questions/32180/testing-if-a-variable-is-empty-in-a-shell-script

Also, https://unix.stackexchange.com/questions/297086/how-to-check-variable-existence-and-compare-with-the-string-in-busybox says

> You'll find that some old Unix shell scripting literature recommend to use [ "x$var" = xproduction ] or [ production = "x$var" ] to do string comparison. That was to overcome bugs in some old versions of the [ and test utility that were confused by some values of $var like !, ( or -n. It should not be needed on modern systems, but something to bear in mind for very old systems.